lib-object-oriented/line.js

```diff
@@ -1,34 +1,77 @@
 import Tag from './tag'
 
 export default class Line {
-  constructor( content ) {
+  constructor({ content, type }) {
     this.content = content
+    this.tagType = type
+    this.symbolCount = 0
+    this.getSymbolCount()
```

Comment:
I usually avoid calling functions in the constructor - it feels side effecty.  As a user of this object, I would need to know how all of these functions work internally in order to be able to effectively consume this object.  As well, it seems as if calling `this.getSymbolCount()` again would increase the size of `this.symbolCount`, which would be undesirable.  

Since `parse()` is the function doing all of the work for `Line`, I would consider deferring these calculations to that function (if a user of this class never calls `parse`, we would presumably not need to perform these calculations anyways).